src/github.com/getlantern/errlog/errlog_test.go

```diff
@@ -53,22 +53,49 @@ func TestAnonymousError(t *testing.T) {
 	rr := &rawReporter{}
 	ReportTo(rr)
 	l := ErrorLoggerFor("my-proxy-package")
-	l.Log(errors.New("any error"))
-	expected := Error{
-		"my-proxy-package",
-		"*errors.errorString",
-		"any error",
-		"",
-		map[string]string{},
-		l.systemInfo,
-		nil,
-		nil,
-		nil,
+	err := errors.New("any error")
+	l.Log(err)
+	expected := &Error{
+		Source:     err,
+		GoPackage:  "my-proxy-package",
+		GoType:     "*errors.errorString",
+		Desc:       "any error",
+		Op:         "",
+		Extra:      map[string]string{},
+		UserLocale: userLocale,
 	}
-	assert.Equal(t, expected, *rr.err, "should log errors created by errors.New")
+	assert.Equal(t, expected, rr.err, "should log errors created by errors.New")
 
 	l.Log(fmt.Errorf("any error"))
-	assert.Equal(t, expected, *rr.err, "should log errors created by fmt.Errorf")
+	assert.Equal(t, expected, rr.err, "should log errors created by fmt.Errorf")
+}
+
+func TestStructuredError(t *testing.T) {
+	rr := &rawReporter{}
+	ReportTo(rr)
+	l := ErrorLoggerFor("my-proxy-package")
+	source := errors.New("any error")
+	err := &Error{
+		Source:    source,
+		GoPackage: "specified package",
+		GoType:    "specified type",
+		Desc:      "specified desc",
+		Op:        "specified op",
+		Extra: map[string]string{
+			"extra1": "specified extra1",
+		},
+	}
+	l.Log(err)
```

Comment:
Here is an example of logging a structured error.  In real usage, one could just do this in one line, and omit the fields that one doesn't care about, for example:

```
l.Log(errlog.Error{
    Source: originalError,
    Op: "proxying",
    Extra: map[string]string{
        "server": "127.0.0.1:443",
    },
})
```

The cool thing about modeling it this way too is that these structured errors can be propagated, since they're real errors, i.e.:

```
err := doSomething()
if err != nil {
    return errlog.Error{
        Source: originalError,
        Op: "proxying",
        Extra: map[string]string{
            "server": "127.0.0.1:443",
        },
    }
}
```